### PR TITLE
gulp: Add tasks for compile and run from src.

### PR DESCRIPTION
**Pull request by vsvipul**
**What's this PR do?**

**Any background context you want to provide?**

**Screenshots?**

**You have tested this PR on:**
  - [ ] Windows
  - [x] Linux/Ubuntu
  - [ ] macOS

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -5,6 +5,27 @@ const electron = require('electron-connect').server.create({
...
+const ts = require("gulp-typescript");
+const tsProject = ts.createProject("tsconfig.json");
+const paths = {
+    files: [
+        'app/**/*',
+        '!app/**/*.ts'
+    ]
+};
+
+gulp.task("copy-non-ts", function () {
```
Comment (priyank-p) [inline gulpfile.js]: We don't need to do this here, since we plan to move the regular `tsc` to output to `src`.

Comment (priyank-p) [reply gulpfile.js]: We can manually remove already outputted files by doing `git clean -xf`.

**Review by priyank-p — APPROVED** [review]
@vsvipul this LGTM! Haven't tried out locally yet though.

File: gulpfile.js (hunk 1 above)
Comment (vsvipul) [reply gulpfile.js]: But, doesn't ```tsc``` just compile .ts to .js . This is for other files like css and html and images.

Comment (priyank-p) [reply gulpfile.js]: Ah, ok makes sense!

File: package.json [hunk 2]
```diff
@@ -31,7 +31,8 @@
     "lint-js": "xo",
     "test": "npm run lint-html && npm run lint-css && npm run lint-js",
     "test-e2e": "gulp test-e2e",
-    "dev": "gulp dev & nodemon --watch app/main --watch app/renderer --exec 'npm test' -e html,css,js",
+    "compile": "gulp compile",
+    "dev": "gulp dev & nodemon --watch app/main --watch app/renderer --exec 'npm test' ",
```
Comment (akashnimare) [inline package.json]: A gentle reminder that `dev` script was broken when I last checked.

**Comment by vsvipul** [comment]
@priyank-p Made some changes. Review please.

File: package.json (hunk 2 above)
Comment (vsvipul) [reply package.json]: Fixed this. I think the nodemon part was not required. The processes are automatically reloaded from the gulp scripts in gulpfile.js .

**Comment by priyank-p** [comment]
@vsvipul I tweaked this to make sure that we run `tsc` if there are files left to compiles otherwise the app throws an error. Thanks, @vsvipul!

**Pull request merged**